Tests/ZIPFoundationTests/ZIPFoundationWritingTests.swift

```diff
@@ -329,4 +329,100 @@ extension ZIPFoundationTests {
         let nonUpdatableArchive = Archive(url: nonUpdatableArchiveURL, accessMode: .update)
         XCTAssertNil(nonUpdatableArchive)
     }
+
+    func testUniqueTemporaryDirectoryURL() {
+        let archive = self.archive(for: #function, mode: .create)
+        var tempURLs = Set<URL>()
+        defer {
+            for url in tempURLs {
+                try? FileManager.default.removeItem(at: url)
+            }
+        }
+        // We choose 2000 temp directories to test workaround for http://openradar.appspot.com/50553219
+        for _ in 1...2000 {
+            autoreleasepool {
```

Comment:
This is the line that currently breaks the Linux build.  
Do we need the ARP here?

Reply:
Without it the memory usage grows since we are in a tight loop.

Reply:
I just looked into that. No I see your point: The memory grows to several gigabytes in this loop 😱. I expected it to grow to some megabytes. 
![Screenshot 2019-05-14 at 22 48 59](https://user-images.githubusercontent.com/1577319/57731249-8c284000-769a-11e9-96c5-428e31b8db26.png)
Instruments shows that a private `NSFileManager` method (`_URLForReplacingItemAtURL`) is allocating 10000s of 1kb blocks for some reason. 
Will have to further investigate what is going on here. This feels wrong.

Reply:
I looked further into this.  
`FileManager.url(for:)` is implemented since Swift 5.0 in corelibs Foundation. Post 5.0 - in the current `master` -  there also have been some refinements to make the method behave more like the Darwin implementation.  
So we have to find workarounds for the following limitations on non-Darwin paltforms: 
 - we can't use `autoreleasepool` 
 - Swift 4.0, 4.1 & 4.2 need fallbacks because `url(for:)` is missing (e.g. by always returning a URL based on `NSTemporaryDirectory()` from `uniqueTemporaryDirectory`)

Reply:
*  I've removed the `autoreleasepool`... it's only a test anyway.
* I've added a condition for non darwin systems for `url(for:)` compatibility.

Reply:
I'm not sure what you'd like to do about the listing failures. They are about file and method length, and that's a bigger task than I care to deal with.